pkg/agent/core/ngt/service/ngt.go

```diff
@@ -216,12 +213,17 @@ func New(cfg *config.NGT, opts ...Option) (nn NGT, err error) {
 // which indicates that the user has NOT been using CoW mode and the index directory is not migrated yet.
 func migrate(ctx context.Context, path string) (err error) {
 	// check if migration is required
+	if !file.Exists(path) {
+		log.Infof("the path %v does not exist. no need to migrate since it's probably the initial state", path)
+		return nil
+	}
 	files, err := file.ListInDir(path)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to list in %v: %w", path, err)
```

Comment:
In vald it would be better to define new error at internal/errors and use it.

Reply:
[690abc7](https://github.com/vdaas/vald/pull/2060/commits/690abc7cfcb09dad8d73693358ecf52a036c7534)